packages/js-sdk/src/sandbox.ts

```diff
@@ -170,149 +264,253 @@ export class Sandbox extends SandboxBase {
+  private readonly url: URL;
+  readonly password: string;
+  private vncCommand: string = "";
+  private readonly novncCommand: string;
+  private readonly desktop: Desktop;
+
+  constructor(desktop: Desktop) {
+    this.desktop = desktop;
+    this.url = new URL(`https://${desktop.getHost(desktop.streamPort)}/vnc.html`);
+    this.password = generateRandomString();
+
+    this.novncCommand = (
+      `cd /opt/noVNC/utils && ./novnc_proxy --vnc localhost:${desktop.vncPort} ` +
+      `--listen ${desktop.streamPort} --web /opt/noVNC > /tmp/novnc.log 2>&1`
+    );
   }
 
   /**
-   * Move the mouse to the given coordinates.
-   * @param x - The x coordinate.
-   * @param y - The y coordinate.
+   * Set the VNC command to start the VNC server.
    */
-  async moveMouse(x: number, y: number) {
-    return this.runPyautoguiCode(`pyautogui.moveTo(${x}, ${y})`)
+  private async setVncCommand(): Promise<void> {
+    let pwdFlag = "-nopw";
+    if (this.desktop.novncAuthEnabled) {
+      await this.desktop.commands.run("mkdir ~/.vnc");
+      await this.desktop.commands.run(`x11vnc -storepasswd ${this.password} ~/.vnc/passwd`);
+      pwdFlag = "-usepw";
+    }
+
+    this.vncCommand = (
+      `x11vnc -display ${this.desktop.display} -forever -wait 50 -shared ` +
+      `-rfbport ${this.desktop.vncPort} ${pwdFlag} 2>/tmp/x11vnc_stderr.log`
+    );
   }
 
-  /**
-   * Get the current mouse position.
-   * @returns An object with `x` and `y` coordinates.
-   */
-  async getCursorPosition() {
-    // We save the value to a file because stdout contains warnings about Xauthority.
-    await this.runPyautoguiCode(`
-x, y = pyautogui.position()
-with open("/tmp/cursor_position.txt", "w") as f:
-    f.write(str(x) + " " + str(y))
-`)
-    // pos is like this: 100 200
-    const pos = await this.files.read('/tmp/cursor_position.txt')
-    const [x, y] = pos.split(' ').map(Number)
-    return { x, y }
+  private async waitForPort(port: number): Promise<boolean> {
+    return await this.desktop.waitAndVerify(
+      `netstat -tuln | grep ":${port} "`, (r: CommandResult) => r.stdout.trim() !== ""
+    );
   }
 
   /**
-   * Get the current screen size.
-   * @returns An object with `width` and `height` properties.
-   */
-  async getScreenSize() {
-    // We save the value to a file because stdout contains warnings about Xauthority.
-    await this.runPyautoguiCode(`
-width, height = pyautogui.size()
-with open("/tmp/size.txt", "w") as f:
-    f.write(str(width) + " " + str(height))
-`)
-    // Size is like this: 100 200
-    const size = await this.files.read('/tmp/size.txt')
-    const [width, height] = size.split(' ').map(Number)
-    return { width, height }
+   * Get the URL to connect to the VNC server.
+   * @param autoConnect - Whether to automatically connect to the server after opening the URL.
+   * @returns The URL to connect to the VNC server.
+   */
+  public getUrl(autoConnect: boolean = true): string {
```

Comment:
should be getUrl({ autoConnect: true, password: desktop.stream.getPassport() })

Reply:
I don't think so. The password is really just an extra security token that is added to the URL and it doesn't provide much utility right now to be accessed via the SDK.